largest-product-array/largest_product_array.py

```diff
@@ -1,20 +1,22 @@
-def largest_product(list):
-    
-    def largest_product(list):
-        product = 0
-        index = 0
-        for x in list:
-            step = list[index[0]] * list[index[1]]
-            if step > product:
-                product = step
-            step = list[index[0]] * list[index + 1[0]]
-            if step > product:
-                product = step
-            step = list[index[1]] * list[index + 1[1]]     
-            if step > product:
-                product = step
-            index += 1
-        return product      
 
-if __name__ == '__main__':
-    largest_product
\ No newline at end of file
+def largest_product(list_in):
+    product = 0
+    index = 0
+    max_index = -1
+    for x in list_in:
+        max_index += 1
+
+    for x in list_in:
+        step_1 = list_in[index][0] * list_in[index][1]
```

Comment:
This assumes that the given matrix is a 2xN matrix. Your code should work for any NxM matrix.